modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperimentServiceImpl.java

```diff
@@ -220,20 +218,19 @@ public SegmentsExperiment runSegmentsExperiment(
 
 		_checkPermissions(segmentsExperiment, ActionKeys.UPDATE);
 
-		Set<Map.Entry<String, Double>> segmentsExperienceKeySplits =
-			segmentsExperienceKeySplitMap.entrySet();
+		Map<Long, Double> segmentsExperienceIdSplitMap = new HashMap<>();
 
-		Stream<Map.Entry<String, Double>> segmentsExperienceKeySplitsStream =
-			segmentsExperienceKeySplits.stream();
+		for (Map.Entry<String, Double> segmentsExperienceKeySplit :
```

Comment:
inline segmentsExperienceKeySplitMap.entrySet().
But for this one, if we inline this in the for loop, don't we also have a performance loss if we need to iterate the for loop many times?

Reply:
@jeffwu0724 What do you mean by the loss? There's no difference between iterating on a variable and iterating directly on `map.entrySet()`.

Reply:
@dantewang 
My confusion is, if we iterate directly on `map.entrySet()`, don't we call this method `map.entrySet()` several times in the loop? While if we store it in a variable, we only call `map.entrySet()` once while declaring.
Thanks for your answer in advance!

Reply:
@jeffwu0724 It doesn't. The for-each loop is internally equivalent with

```
Iterator iterator = map.entrySet().iterator();

while (iterator.hasNext()) {
    Map.Entry entry = iterator.next();
    ......
}
```

One of my earliest tasks on Liferay Portal was to convert the above while-loop with foreach loop wherever possible :P

Reply:
Thanks @dantewang for your detailed explanation!